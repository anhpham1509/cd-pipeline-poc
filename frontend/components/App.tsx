import React from "react"
import {hot} from "react-hot-loader"
import "./App.scss"

interface Props {
  show: boolean
}

interface State {
}

export class App extends React.Component<Props, State> {
  public render() {
    const {show} = this.props;
    return (
      <div className="app">
        {show && <h1>Hello World!</h1>}
      </div>
    )
  }
}

export default hot(module)(App)
