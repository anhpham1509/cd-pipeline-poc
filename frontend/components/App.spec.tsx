import React from "react"
import enzyme from "enzyme"

import {App} from "./App"

describe("<App>", () => {
  it("should render Hello World!", () => {
    const app = enzyme.render(<App show={false}/>)
    expect(app).toMatchSnapshot()
  })
})
