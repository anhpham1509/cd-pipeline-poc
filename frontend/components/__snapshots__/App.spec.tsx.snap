// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should render Hello World! 1`] = `
<div
  class="app"
/>
`;
